sql/parse/parse.go

```diff
@@ -3179,6 +3179,31 @@ func ExprToExpression(ctx *sql.Context, e sqlparser.Expr) (sql.Expression, error
 	}
 }
 
+// handleCollateExpr is meant to handle generic text-returning expressions that should be reinterpreted as a different collation.
+func handleCollateExpr(ctx *sql.Context, charSet sql.CharacterSetID, expr *sqlparser.CollateExpr) (sql.Expression, error) {
+	//TODO: rename this from Charset to Collation
+	collation, err := sql.ParseCollation(nil, &expr.Charset, false)
+	if err != nil {
+		return nil, err
+	}
+	if collation.Sorter() == nil {
```

Comment:
This is kind of an odd way to check if the collation is unsupported. Maybe check the ID against a public Unsupported const?